cloudinit/sources/DataSourceAzure.py

```diff
@@ -890,36 +891,20 @@ def _create_report_ready_marker(self):
         )
 
     @azure_ds_telemetry_reporter
-    def _report_ready_if_needed(self):
-        """Report ready to the platform if the marker file is not present,
-        and create the marker file.
+    def _report_ready_for_pps(self, lease: dict) -> None:
+        """Report ready for PPS, creating the marker file upon completion.
+
+        :raises sources.InvalidMetaDataException: On error reporting ready.
+
+        :return: Name of interface used to report ready.
```

Comment:
this function doesn't return anything though?

Reply:
Ah yes, old approach. Removed

Reply:
thanks :D